integration/duckdb_lance/src/arrow.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright 2023 Lance Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Arrow / DuckDB conversion.
+
+use std::{collections::HashMap, ffi::CString};
+
+use arrow_array::{
+    cast::{as_boolean_array, as_primitive_array},
+    types::*,
+    Array, ArrowPrimitiveType, BooleanArray, PrimitiveArray, RecordBatch, StringArray,
+};
+use arrow_schema::DataType;
+use duckdb_extension_framework::{duckly::idx_t, DataChunk, LogicalType, LogicalTypeId, Vector};
+
+use crate::{Error, Result};
+
+pub fn to_duckdb_type_id(data_type: &DataType) -> Result<LogicalTypeId> {
+    use LogicalTypeId::*;
+
+    let type_id = match data_type {
```

Comment:
duckdb's arrow integration has a mapping for this. just double checking that the two match